include/deal.II/base/exception_macros.h

```diff
@@ -870,39 +870,33 @@
  * This macro is first used in step-7, where we show another example of
  * a context where it is frequently used.
  */
-#define DEAL_II_ASSERT_UNREACHABLE()                             \
-  ::dealii::deal_II_exceptions::internals::issue_error_noreturn( \
-    ::dealii::deal_II_exceptions::internals::ExceptionHandling:: \
-      abort_or_throw_on_exception,                               \
-    __FILE__,                                                    \
-    __LINE__,                                                    \
-    __PRETTY_FUNCTION__,                                         \
-    nullptr,                                                     \
-    nullptr,                                                     \
-    ::dealii::StandardExceptions::ExcMessage(                    \
```

Comment:
If I read this patch right, then all you do is replace `issue_error_noreturn()` here by `do_unreachable()`. But then why could the split between host and device code not have gone right into `issue_error_noreturn()`? That would have fixed all of the other places where we call that function.

Reply:
Making that work is more difficult, because we construct an ``ExcMessage`` in the code above, which is not annotated to work on device. I think one would need to modify all exception classes and functions to make that work.

Reply:
I see, thanks for the explanation. (So much for the first G in GPGPU...)